File: data/src/main/java/com/crhonvas/data/remote/api/posts/IPostsApi.java
package com.crhonvas.data.remote.api.posts;

import com.crhonvas.data.remote.model.comments.CommentApiResponse;
import com.crhonvas.data.remote.model.posts.PostsApiResponse;

import java.util.List;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface IPostsApi {
    @GET("posts")
    Single<List<PostsApiResponse>> getPosts();

    @GET("posts/{postId}")
    Single<PostsApiResponse> getPost(@Path("postId") Integer postId);
}

File: data/src/main/java/com/crhonvas/data/remote/api/posts/IPostsApi.kt
package com.crhonvas.data.remote.api.posts

import com.crhonvas.data.remote.model.posts.PostsApiResponse
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path

interface IPostsApi {
    @get:GET("posts")
    val posts: Single<List<PostsApiResponse>>

    @GET("posts/{postId}")
    fun getPost(@Path("postId") postId: Int?): Single<PostsApiResponse>
}